packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -232,6 +232,10 @@ export default class Extension {
     const { request, resolve } = queued;
 
     this.#state.saveMetadata(request);
+    request.metaCalls &&
```

Comment:
This should not happen on save. It should kick in for the instance where the signing takes place. 

The issue you have here is that each update injects their own and they are different.

Reply:
Thanks, @jacogr  for your fast feedback. Moved to the signing method.